### ISSUE TITLE
Sampler for valid dataloader

### ISSUE THREAD

**Issue by chunhanl** [AnnaAraslanova/FBNet#2]
In general_functions/dataloaders.py  

Shouldn't the sampler of the val_loader be valid_sampler ?
val_loader = torch.utils.data.DataLoader(
        train_data, batch_size=batch_size, sampler=**valid_sampler**,
        pin_memory=True, num_workers=16)

**Comment by Catosine** [comment]
Hello @chunhanl ! I think this is due to the usage of get_loaders().

When **searching** for architecture, the valid dataset is only used to optimize model structure, and all related accuracy does not suggest the retrain performance. The only goal is to find the best model structures, not the best model performance.  So at this point, we would expect the datasets are processed the same as the training data during retrain.

And if you checked codes about retrain, you would notice that get_loaders() is only used to get train data, and the validation data is received by get_test_loader().

GL,

**Comment by chunhanl** [comment]
Hello @Catosine 
Thank you for your response and information.

I think that the author was intending to split the training data into two subsets: one for weights and one for arch ( train_w_loader, train_thetas_loader = get_loaders(....)  @ supernet_main_file.py)
which two of the data should not overlap. Also the valid_sampler is declared but not used in getloaders(), which I believe it might be a typo mistake.

Best,

**Comment by Catosine** [comment]
@chunhanl You are right! Thanks:)

**Comment by AnnaAraslanova** [comment]
I fixed the line in the code.  Thank you
